pandas/tests/copy_view/test_indexing.py

```diff
@@ -0,0 +1,538 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+import pandas._testing as tm
+import pandas.core.common as com
+
+# -----------------------------------------------------------------------------
+# Indexing operations taking subset + modifying the subset/parent
+
+
+def test_subset_column_selection(using_copy_on_write):
+    # Case: taking a subset of the columns of a DataFrame
+    # + afterwards modifying the subset
+    df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [0.1, 0.2, 0.3]})
+    df_orig = df.copy()
+
+    subset = df[["a", "c"]]
+
+    if using_copy_on_write:
+        # the subset shares memory ...
+        assert np.may_share_memory(subset["a"].values, df["a"].values)
+        # ... but uses CoW when being modified
+        subset.iloc[0, 0] = 0
+    else:
+        assert not np.may_share_memory(subset["a"].values, df["a"].values)
+        # INFO this no longer raise warning since pandas 1.4
+        # with pd.option_context("chained_assignment", "warn"):
+        #     with tm.assert_produces_warning(com.SettingWithCopyWarning):
+        subset.iloc[0, 0] = 0
+
+    assert not np.may_share_memory(subset["a"].values, df["a"].values)
```

Comment:
any particular reason for may_share_memory instead of shares_memory?

Reply:
No particular reason except for ignorance, I suppose I copied it from somewhere. We also have a `tm.shares_memory`, I see, so I could also directly use that instead of `np.shares_memory`?

Reply:
Either way.  i'd go with np.shares_memory iff everything being checked is an ndarray

Reply:
Updated to use `np.shares_memory`